### PR TITLE
Allow strings in layout list

### LINKED ISSUE #2890: [BUG] No Layout Exception when using string elements when the layout is a list
In all versions of Dash this works:

```python 
app.layout = html.Div(["Select City", dcc.Dropdown()])
```

In dash >= 2.17, this will throw an error:

```python 
app.layout = ["Select City", dcc.Dropdown()]
```

Here is a full example.  When I run the app, I see the following error.  After refreshing the screen it works fine.

```python
from dash import Dash, dcc
app = Dash(__name__)

app.layout = ["Select City", dcc.Dropdown()]

if __name__ == '__main__':
    app.run(debug=True)


```

image

### PR DESCRIPTION
**Pull request by leeagustin**
Fixes #2890.

The problem was that strings were not allowed when the page layout was defined as a list. The issue only happens the first time—and not the second one after a refresh—because the layout validation using `validate_layout()` is only done the first time the page is accessed, resulting in the error `dash.exceptions.NoLayoutException: List of components as layout must be a list of components only.`

The solution is simply to allow strings to be included when the page layout is a list.

## Contributor Checklist

- [x] I have broken down my PR scope into the following TODO tasks
   -  [x] Allows strings if layout is a list
- [x] I have run the tests locally and they passed. (refer to testing section in [contributing](https://github.com/plotly/dash/blob/master/CONTRIBUTING.md))
- [x] I have added tests, or extended existing tests, to cover any new features or bugs fixed in this PR

### optionals

- [x] I have added entry in the `CHANGELOG.md`
- [ ] If this PR needs a follow-up in **dash docs**, **community thread**, I have mentioned the relevant URLS as follows
    -  [ ] this GitHub [#PR number]() updates the dash docs
    -  [ ] here is the show and tell thread in Plotly Dash community

### CODE REVIEW

File: dash/_validate.py [hunk 1]
```diff
@@ -439,11 +439,13 @@ def _validate_id(comp):
...
     if isinstance(layout_value, (list, tuple)):
         for component in layout_value:
+            if isinstance(component, (str,)):
+                continue
             if isinstance(component, (Component,)):
                 _validate(component)
             else:
                 raise exceptions.NoLayoutException(
-                    "List of components as layout must be a list of components only."
+                    "List of components as layout must be a list of strings and components only."
```
Comment (T4rk1n) [inline dash/_validate.py:448]:
```suggestion
                    "Only strings and components are allowed in a list layout."
```

**Review by T4rk1n — COMMENTED** [review]
Looks good just missing a changelog entry.

File: dash/_validate.py (hunk 1 above)
Comment (leeagustin) [reply dash/_validate.py]: Modified in d9d963f

**Comment by leeagustin** [comment]
> Looks good just missing a changelog entry.

Added in ba30a4d

**Review by T4rk1n — APPROVED** [review]
:woman_dancing: Thank you

**Comment by leeagustin** [comment]
It seems that one of the tests failed (`tests/integration/renderer/test_persistence.py:209 test_rdps005_persisted_props`), although these changes shouldn't have influenced it. The test is also passing on my machine. Is this what's blocking the merge?

**Comment by leeagustin** [comment]
> 💃 Thank you

No problem! 💯 

It seems that one of the tests failed (tests/integration/renderer/test_persistence.py:209 test_rdps005_persisted_props), although these changes shouldn't have influenced it. The test is also passing on my machine. Is this what's blocking the merge?

**Comment by leeagustin** [comment]
No problem @T4rk1n! Thank you for reviewing my changes! 🙏 

It seems that one of the tests failed (tests/integration/renderer/test_persistence.py:209 test_rdps005_persisted_props), although these changes shouldn't have influenced it. The test is also passing on my machine. Is this what's blocking the merge?

**Comment by leeagustin** [comment]
Ah okay, sounds good. Thanks for the clarification!

**Comment by ndrezn** [comment]
Nice fix!

**Comment by leeagustin** [comment]
Thank you @ndrezn! 🙏

**Comment by T4rk1n** [comment]
> test_rdps005_persisted_props

Think there is something wrong with the CI image/container because that test fails consistently on Python 3.8 image. I will merge the PR in another branch then in dev just to be sure.

**Pull request merged**